test-utils/state-viewer/src/main.rs

```diff
@@ -35,11 +44,38 @@ fn load_trie(
         near_config.client_config.tracked_shards.clone(),
     );
     let head = chain_store.head().unwrap();
-    let last_block = chain_store.get_block(&head.last_block_hash).unwrap().clone();
-    let mut state_roots = vec![];
-    for chunk in last_block.chunks.iter() {
-        state_roots.push(chunk.inner.prev_state_root.clone());
-    }
+    let last_block = match stop_height {
+        Some(height) => {
+            // find the first final block whose height is at least `height`.
+            let mut cur_height = height + 1;
+            loop {
+                if cur_height >= head.height {
+                    panic!("No final block with height >= {} exists", height);
+                }
+                let cur_block_hash = match chain_store.get_block_hash_by_height(cur_height) {
+                    Ok(hash) => hash,
+                    Err(_) => {
+                        cur_height += 1;
+                        continue;
+                    }
+                };
+                let last_final_block_hash = chain_store
+                    .get_block_header(&cur_block_hash)
+                    .unwrap()
+                    .inner_rest
+                    .last_quorum_pre_commit;
+                let last_final_block = chain_store.get_block(&last_final_block_hash).unwrap();
+                if last_final_block.header.inner_lite.height >= height {
+                    break last_final_block.clone();
+                } else {
+                    cur_height += 1;
+                    continue;
```

Comment:
i don't think you need a continue here

Reply:
I don't understand. It wouldn't compile without `continue`.  What do you suggest instead?